Simplenote/src/main/java/com/automattic/simplenote/viewmodels/CompleteMagicLinkViewModel.kt

```diff
@@ -0,0 +1,62 @@
+package com.automattic.simplenote.viewmodels
+
+import androidx.annotation.StringRes
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.automattic.simplenote.R
+import com.automattic.simplenote.Simplenote
+import com.automattic.simplenote.di.IO_THREAD
+import com.automattic.simplenote.networking.SimpleHttp
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.launch
+import org.json.JSONObject
+import java.io.IOException
+import javax.inject.Inject
+import javax.inject.Named
+
+@HiltViewModel
+class CompleteMagicLinkViewModel @Inject constructor(
+    private val simplenote: Simplenote,
+    private val simpleHttp: SimpleHttp,
+    @Named(IO_THREAD) private val ioDispatcher: CoroutineDispatcher,
+    ) : ViewModel() {
+    private val _magicLinkUiState = MutableLiveData<MagicLinkUiState>()
+    val magicLinkUiState: LiveData<MagicLinkUiState> get() = _magicLinkUiState
+
+    fun completeLogin(authKey: String, authCode: String) = viewModelScope.launch(ioDispatcher) {
+        _magicLinkUiState.postValue(MagicLinkUiState.Loading(messageRes = R.string.magic_link_complete_login_loading_message))
+        try {
+            simpleHttp.firePostRequest(
+                "account/complete-login",
+                mapOf(Pair("auth_key", authKey), Pair("auth_code", authCode))
+            ).use { response ->
```

Comment:
Consider using a repository for this kind of call. Maybe you can create a `MagicLinkRepository` to handle the calls and then inject it into the view models that need it.